src/app/core/modals/add-source-modal/add-source-modal.component.ts

```diff
@@ -104,6 +107,13 @@ export class AddSourceModalComponent extends DirectiveBase {
     localFoldersToAdd                                       = [];
     closedProjectOptions: { value: string, text: string }[] = null;
```

Comment:
Add typehint for `testAttr` here because you're adding it to the object on line 133. Better yet, extract `{value: string ... }` into its own type and typehint the parameter for `projectDropdownRenderOptions`